File: Kadane'sAlgorithm/Kadane.py
#Author: Khushboo Gupta
#Implementation of Kadane's Algorithm in Python
#time Complexity: O(n)

#code is as follows:

from sys import maxint 
def maxSubArraySum(a,size): 
       
    max_so_far = -maxint + 1
    max_ending_here = 0
       
    for i in range(0, size): 
        max_ending_here = max_ending_here + a[i] 
        if (max_so_far < max_ending_here): 
            max_so_far = max_ending_here 
  
        if max_ending_here < 0: 
            max_ending_here = 0   
    return max_so_far 

File: Kadane'sAlgorithm/Kadane.go
package main

import (
	"fmt"
)

func maxSubarraySum(arr []int) int {
	if len(arr) == 0 {
		return 0
	}

	maxEndingHere := arr[0]
	maxSoFar := arr[0]

	for i := 1; i < len(arr); i++ {
		maxEndingHere = max(arr[i], maxEndingHere+arr[i])
		maxSoFar = max(maxSoFar, maxEndingHere)
	}

	return maxSoFar
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func main() {
	arr := []int{-2, 1, -3, 4, -1, 2, 1, -5, 4}
	maxSum := maxSubarraySum(arr)
	fmt.Println("Maximum subarray sum:", maxSum)
}
